addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/CORSActiveScanRule.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+
+/**
+ * The CORS active scan rule identifies Cross-Origin Resource Sharing (CORS) support and overly
+ * lenient or buggy implementations
+ *
+ * @author CravateRouge
+ */
+public class CORSActiveScanRule extends AbstractAppPlugin {
+    private static Logger LOG = Logger.getLogger(CORSActiveScanRule.class);
+    private static final String RANDOM_NAME = RandomStringUtils.random(8, true, true);
+
+    @Override
+    public void scan() {
+        HttpMessage baseMsg = getBaseMsg();
+        URI uri = baseMsg.getRequestHeader().getURI();
+        String authority = uri.getEscapedAuthority();
+        String scheme = uri.getScheme();
+        String handyScheme = scheme + "://";
+
+        // Order of likelihood and severity
+        String[] payloads = {
+            handyScheme + RANDOM_NAME + ".com",
+            "null",
+            handyScheme + RANDOM_NAME + "." + authority,
+            handyScheme + authority + "." + RANDOM_NAME + ".com",
+            // URL encoded backtick used to bypass weak Regex matching only alphanumeric chars to
+            // validate the domain: https://www.corben.io/tricky-CORS/
+            handyScheme + authority + "%60" + RANDOM_NAME + ".com",
+            null,
+            handyScheme + authority
+        };
+
+        boolean secScheme = false;
+        if (scheme == "https") {
+            secScheme = true;
+            payloads[5] = "http://" + authority;
+        }
+
+        for (String payload : payloads) {
+            HttpMessage msg = getNewMsg();
+            msg.getRequestHeader().setHeader("Origin", payload);
+            try {
+                sendAndReceive(msg);
+                String acaoKey = HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN;
+                String acao = msg.getResponseHeader().getHeader(acaoKey);
+                // If there is an ACAO header an alert will be triggered
+                if (acao == null) continue;

```

Comment:
While this is syntactically correct it would be best to still use curly braces here for consistency.